widgets/src/main/java/org/pentaho/gwt/widgets/client/filechooser/FileChooserDialog.java

```diff
@@ -402,20 +402,13 @@ public void setSubmitOnEnter( boolean submitOnEnter ) {
   }
 
   @Override
+  @Deprecated
   public boolean onKeyDownPreview( char key, int modifiers ) {
     // Use the popup's key preview hooks to close the dialog when either
     // enter or escape is pressed.
-    switch ( key ) {
-      case KeyboardListener.KEY_ENTER:
-        if ( isSubmitOnEnter() ) {
-          onOk();
-        }
-        break;
-      case KeyboardListener.KEY_ESCAPE:
-        onCancel();
-        break;
+    if ( key == (char) KeyboardListener.KEY_ESCAPE ) {
```

Comment:
Please add a deprecation annotation and comment on the "submit on enter" methods and remove the field. The same would apply to the FileChooser's one methods and field.

Also, we should remove the uses of these methods by the Pentaho Platform project and remove/deprecate anything on that end as well.

Reply:
We discussed offline and I was wrong. The "Submit on Enter/Return" behavior is standard accessible behavior for [single-line textbox elements](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/textbox_role#aria-multiline).

As such, removing the dialog's Enter/Return behavior actually takes us farther from the standard behavior.
We agreed to remove this behavior temporarily, on this story, and to handle re-instanting it correctly on a followup story. The Enter/Return key should be handled at the dialog level, in the normal event phase (not in the capture/preview) and should only take effect when the event has not yet been processed (default action not prevented) and when the event was targeted at a textbox (aria role = textbox) (a multi-line textbox will have marked the event as handled).

Reply:
Checking further on few more dialogs, I don't think it's necessary to have Enter key behavior at the dialog level. This unnecessarily overrides the natural behavior of the interactive element and is behaving strangely on Enter (Both the logic of onOk() and the onClickListener for that element are getting executed). I think it's better to completely remove the Enter behavior at the dialog level which is similar to DialogBox.

Reply:
It is precisely to prevent overriding the local behavior that I included the condition «should only take effect when the event has not yet been processed (default action not prevented)».